app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java

```diff
@@ -74,6 +75,27 @@ public void start(Stage primaryStage) throws IOException {
     Thread.currentThread().setUncaughtExceptionHandler(Shuffleboard::uncaughtException);
     onOtherAppStart = () -> Platform.runLater(primaryStage::toFront);
 
+    // Before we load components that only work with Java 8, check to make sure
+    // the application is running on Java 8. If we are running on an invalid
+    // version, show an alert and exit before we get into trouble.
+    String javaSpec = System.getProperty("java.specification.version");
+    String javaVersion = System.getProperty("java.version");
+    if (!"1.8".equals(javaSpec)) {
+      System.out.println(javaSpec);
+      Alert invalidVersionAlert = new Alert(Alert.AlertType.ERROR);
+      invalidVersionAlert.setHeaderText("Invalid JRE Version!");
+      invalidVersionAlert.setContentText(
+          String.format("You are using an unsupported Java version: %s\n"
+                  + "Please download Java 8 and uninstall Java %s.",
```

Comment:
Do they really have to uninstall their other version of java? They just need to make sure that `JAVA_HOME` is set correctly right?

Reply:
Certain jar launchers do not respect the java home environment variable.

Reply:
Then ship this with the Gradle Application plugin shell/command line launcher that will always respect `JAVA_HOME`.

Reply:
Not an option for this year, as the DS runs the jar directly.

Reply:
So have the DS code search for java 8 and try to launch with java 8.

Reply:
Due to the approaching deadline of today, I think we go with this as is and circle back after season is over to make shuffleboard work with both JRE 8 & 9 or make this a better UX.